src/frontend/XAPI/XAPIStatement.ts

```diff
@@ -0,0 +1,144 @@
+  screenSize: 'https://w3id.org/xapi/video/extensions/screen-size',
+  sessionId: 'https://w3id.org/xapi/video/extensions/session-id',
+  speed: 'https://w3id.org/xapi/video/extensions/speed',
+  track: 'https://w3id.org/xapi/video/extensions/track',
+  userAgent: 'https://w3id.org/xapi/video/extensions/user-agent',
+  videoPlaybackSize:
+    'https://w3id.org/xapi/video/extensions/video-playback-size',
+  volume: 'https://w3id.org/xapi/video/extensions/volume',
+};
+
+const ResultExtensionsDefinition: {
+  playedSegment: string;
+  progress: string;
+  time: string;
+  timeFrom: string;
+  timeTo: string;
+} = {
+  playedSegment: 'https://w3id.org/xapi/video/extensions/played-segments',
+  progress: 'https://w3id.og/xapi/video/extensions/progress',
+  time: 'https://w3id.org/xapi/video/extensions/time',
+  timeFrom: 'https://w3id.org/xapi/video/extensions/time-from',
+  timeTo: 'https://w3id.org/xapi/video/extensions/time-to',
+};
+
+export const verbDefinition: {
+  completed: string;
+  initialized: string;
+  interacted: string;
+  paused: string;
+  played: string;
+  seeked: string;
+  terminated: string;
+} = {
+  completed: 'http://adlnet.gov/expapi/verbs/completed',
+  initialized: 'http://adlnet.gov/expapi/verbs/initialized',
+  interacted: 'http://adlnet.gov/expapi/verbs/interacted',
+  paused: 'https://w3id.org/xapi/video/verbs/paused',
+  played: 'https://w3id.org/xapi/video/verbs/played',
+  seeked: 'https://w3id.org/xapi/video/verbs/seeked',
+  terminated: 'http://adlnet.gov/expapi/verbs/terminated',
+};
+
+export class XAPIStatement {
+  constructor(private jwt: string, private sessionId: string) {}
+
+  initialized(contextExtensions: InitializedContextExtensions): void {
+    // find a way to remove this undefined type. There is no undefined value
+    const extensions: {
+      [key: string]: string | boolean | number | undefined;
+    } = {};
+    for (const key of Object.keys(contextExtensions)) {
+      extensions[ContextExtensionsDefintion[key as contextExtensionsKey]] =
+        contextExtensions[key as contextExtensionsKey];
+    }
+
+    extensions[ContextExtensionsDefintion.sessionId] = this.sessionId;
+
+    const data = {
+      context: {
+        extensions,
+      },
+      verb: {
+        display: {
+          'en-US': 'initialized',
+        },
+        id: verbDefinition.initialized,
+      },
+    };
+
+    this.send(data);
+  }
+
+  played(resultExtensions: PlayedResultExtensions): void {
+    const data = {
+      context: {
+        extensions: {
+          [ContextExtensionsDefintion.sessionId]: this.sessionId,
+        },
+      },
+      result: {
+        extensions: {
+          [ResultExtensionsDefinition.time]: resultExtensions.time,
+        },
+      },
+      verb: {
+        display: {
+          'en-US': 'played',
+        },
+        id: verbDefinition.played,
+      },
+    };
+
+    this.send(data);
+  }
+
+  private send(data: DataPayload) {
+    data.id = uuid();
```

Comment:
It's probably a better idea to create a new object here rather than mutate the argument